addons/mrp_subcontracting_account/models/stock_picking.py

```diff
@@ -23,3 +23,15 @@ def _prepare_subcontract_mo_vals(self, subcontract_move, bom):
         if bom.product_tmpl_id.cost_method in ('fifo', 'average'):
             vals = dict(vals, extra_cost=subcontract_move._get_price_unit())
         return vals
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    def _prepare_common_svl_vals(self):
+        # if we are subcontracting we want to have the receipt move
+        # on the svl, not the MO
+        vals = super()._prepare_common_svl_vals()
+        if self.move_dest_ids.route_ids and self.move_dest_ids.warehouse_id.subcontracting_route_id.id in self.move_dest_ids.route_ids.ids:
```

Comment:
```suggestion
        if self.move_dest_ids.filtered(lambda m: m.is_subcontracted):
            vals['stock_move_id'] = self.move_dest_ids.id
```

I don't think it's possible to have more than one move dest in this case but we could filter and select the first to be sure